src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/SubmissionComponent.scala

```diff
@@ -246,9 +245,27 @@ trait SubmissionComponent {
     }
 
     def loadSubmissionWorkflows(submissionId: UUID): ReadAction[Seq[Workflow]] = {
-      workflowQuery.findWorkflowsBySubmissionId(submissionId).result.flatMap(recs => DBIO.sequence(recs.map { rec =>
-        workflowQuery.get(rec.id) map(wf => wf.get)
-      }))
+      // workflow left join entity left outer join workflow_message
+      val workflows = for {
+        ((workflow, entity), message) <-
+          workflowQuery.filter(_.submissionId===submissionId) join entityQuery on (_.workflowEntityId === _.id) joinLeft workflowMessageQuery on (_._1.id === _.workflowId)
+      } yield (workflow, entity, message)
+
+      workflows.result.map{recs =>
+        recs.groupBy(_._1.id).map{x =>
+          val wr:WorkflowRecord = x._2.head._1
```

Comment:
personally I get very confused by all the tuples (which may just be lack of slick experience), but comments would be great to know what some of this is doing.  Something like val wr:WorkflowRecord = x._2.head._1 is at least not too bad because the value shows what that means, but I don't quite know what x is or the x._2 is that gets mapped below to get y._3